core/src/main/java/io/grpc/inprocess/InProcessServerBuilder.java

```diff
@@ -28,6 +28,47 @@
  * its name.
  *
  * <p>The server is intended to be fully-featured, high performance, and useful in testing.
+ *
+ * <h3>Using JUnit TestRule</h3>
+ * The class "GrpcServerRule" (from the "grpc-java/testing") is a JUnit TestRule that
+ * creates a {@link InProcessServer} and a {@link io.grpc.ManagedChannel ManagedChannel}. This
+ * test rule contains the boiler plate code shown below. The classes "HelloWorldServerTest" and
+ * "HelloWorldClientTest" (from "grpc-java/examples") demonstrate basic usage.
+ *
+ * <h3>Usage example</h3>
+ * <h4>Server and client channel setup</h4>
+ * <pre>
+ *   MutableHandlerRegistry serviceRegistry = new MutableHandlerRegistry();
+ *   addServicesHelper(serviceRegistry); // your code here
+ *   ExecutorService serverExecutor = Executors.newCachedThreadPool();
+ *   Server server = InProcessServerBuilder.forName("unique-name")
+ *       .executor(serverExecutor)
+ *       .fallbackHandlerRegistry(serviceRegistry)
+ *       .build().start();
+ *   ExecutorService clientExecutor = Executors.newCachedThreadPool();
+ *   ManagedChannel channel = InProcessChannelBuilder.forName("unique-name")
+ *       .executor(clientExecutor)
+ *       .build();
+ * </pre>
+ *
+ * <h4>Client call usage</h4>
+ * <pre>
+ *   MethodDescriptor&lt;Integer, Integer&gt; methodDescriptor =
```

Comment:
For this example, I probably wouldn't bother showing this MethodDescriptor-based code. I'd probably just say something like "it's a channel to use like normal" and then show the `newBlockingStub()` example line.